lib/ide/src/view/text_editor.rs

```diff
@@ -13,25 +16,65 @@ use basegl::display::world::*;
 
 use nalgebra::Vector2;
 use nalgebra::zero;
+use web_sys::KeyboardEvent;
+
 
 
 // ==================
 // === TextEditor ===
 // ==================
 
+shared! { TextEditor
+
 /// TextEditor allows us to edit text files or Enso Modules. Extensible code highlighting is
 /// planned to be implemented for it.
-#[derive(Clone,Debug)]
-pub struct TextEditor {
-    text_field : TextField,
-    padding    : TemporaryPadding,
-    position   : Vector2<f32>,
-    size       : Vector2<f32>
+#[derive(Debug)]
+pub struct TextEditorData {
+    text_field   : TextField,
+    padding      : TemporaryPadding,
+    position     : Vector2<f32>,
+    size         : Vector2<f32>,
+    controller   : Handle,
+    key_listener : Option<KeyboardListener>
 }
 
+impl {
+    /// Updates the underlying display object.
+    pub fn update(&self) {
+        let padding  = self.padding;
+        let position = self.position;
+        let position = Vector3::new(position.x + padding.left, position.y + padding.bottom, 0.0);
```

Comment:
Is the last coordinate always zero in such patterns? We might want to make it a named constant, or if the pattern repeats, have a special function that creates Vector3 from two values and filling last value with zero.

Reply:
We should not use literals in code until they are refactored to variables. Michal's question is very valid.